source/speech/speech.py

```diff
@@ -1147,8 +1152,14 @@ def speakTextInfo(
 	)
 
 	speechGen = GeneratorWithReturn(speechGen)
+	symbolLevel: Optional[characterProcessing.SymbolLevel] = None
+	if unit == textInfos.UNIT_CHARACTER:
+		symbolLevel = characterProcessing.SymbolLevel.ALL
+	elif unit == textInfos.UNIT_WORD:
+		if config.conf["speech"]["symbolLevelWordAll"]:
+			symbolLevel = characterProcessing.SymbolLevel.ALL
 	for seq in speechGen:
-		speak(seq, priority=priority)
+		speak(seq, symbolLevel=symbolLevel, priority=priority)

```

Comment:
Please disregard my last comment - wrapping the text in the `Symbol` command in `editableText` would not improve the situation at all. In this line you're overwriting symbol level for the entire sequence passed to `speech.speak` which I believe is causing the unnecessary punctuation to be announced. This code should be changed so that  the punctuation level is changed only for the duration of announcing the word / character and not for the entire sequence This would also eliminate the need for the new `UserInterface` command decreasing the complexity so this is a good thing.

Reply:
> This code should be changed so that the punctuation level is changed only for the duration of announcing the word / character and not for the entire sequence.

How would you do that?

Reply:
The responsibility of deciding what symbol level to use should be moved from `speakTextInfo` (this function has not enough information to make a right decision as it receives only a sequence of strings) to the `getTextInfoSpeech`. `getTextInfoSpeech` should retrieve list of all fields and strings  in the textInfo using `getTextWithFields` as it does now, and then should enclose all strings in a speech command indicating to `speech.speak` with which punctuation level the given string should be spoken.